### PR TITLE
DEV: `/channel` -> `/c` chat route rename

### PR DESCRIPTION
**Pull request by romanrizzi**
Also swap the channel id and channel slug params to be consistent with core.

### CODE REVIEW

File: plugins/chat/assets/javascripts/discourse/components/chat-channel-card.hbs [hunk 1]
```diff
@@ -11,7 +11,7 @@
     <div class="chat-channel-card__header">
       <LinkTo
         @route="chat.channel"
-        @models={{array @channel.id (slugify-channel @channel)}}
+        @models={{array (slugify-channel @channel) @channel.id}}
```
Comment (jjaffeux) [inline plugins/chat/assets/javascripts/discourse/components/chat-channel-card.hbs:14]: Im thinking, maybe part of this we introduce a `@channel.slug` so we don't have to call `slugify-channel` everywhere. Given we have  slug in the backend now, this should be easy to do.

File: plugins/chat/plugin.rb [hunk 2]
```diff
@@ -662,6 +656,27 @@ def self.onebox_template
     post "/:chat_channel_id" => "chat#create_message"
     put "/flag" => "chat#flag"
     get "/emojis" => "emojis#index"
+
+    get "/c/:channel_id" => "chat#respond"
```
Comment (jjaffeux) [inline plugins/chat/plugin.rb:660]: Do we still need this?

File: plugins/chat/spec/plugin_spec.rb [hunk 3]
```diff
@@ -144,7 +144,7 @@
       ).chat_message
     end
 
-    let(:chat_url) { "#{Discourse.base_url}/chat/channel/#{chat_channel.id}" }
+    let(:chat_url) { "#{Discourse.base_url}/chat/c/#{chat_channel.id}" }
```
Comment (jjaffeux) [inline plugins/chat/spec/plugin_spec.rb:147]: Why we dont use `/chat/c/-/:id` here?

**Comment by jjaffeux** [comment]
Note that I fixed linting and a failing test @romanrizzi

**Comment by jjaffeux** [comment]
The two failing tests in backend are fixed in main I think.

**Comment by jjaffeux** [comment]
@romanrizzi also fixed the two failures related to `chat/message/:id` transitioning to wrong params

File: plugins/chat/assets/javascripts/discourse/components/chat-channel-card.hbs [hunk 4]
```diff
@@ -11,7 +11,7 @@
     <div class="chat-channel-card__header">
       <LinkTo
         @route="chat.channel"
-        @models={{array @channel.id (slugify-channel @channel)}}
+        @models={{array @channel.slugifiedTitle @channel.id}}
```
Comment (jjaffeux) [inline plugins/chat/assets/javascripts/discourse/components/chat-channel-card.hbs:14]: We could maybe have a `routeModels` methods on `channel` which would return `[slug, id]`? what do you think?

File: plugins/chat/assets/javascripts/discourse/components/chat-channel-row.hbs [hunk 5]
```diff
@@ -1,6 +1,6 @@
 <LinkTo
   @route="chat.channel"
-  @models={{array @channel.id (or @channel.slug "-")}}
+  @models={{array (or @channel.slug "-") @channel.id}}
```
Comment (jjaffeux) [inline plugins/chat/assets/javascripts/discourse/components/chat-channel-row.hbs:3]: the "-" is not needed anymore no?

File: plugins/chat/assets/javascripts/discourse/initializers/chat-sidebar.js [hunk 6]
```diff
@@ -220,7 +220,7 @@ export default {
             }
 
             get models() {
-              return [this.channel.id, slugifyChannel(this.channel)];
+              return [slugifyChannel(this.channel), this.channel.id];
```
Comment (jjaffeux) [inline plugins/chat/assets/javascripts/discourse/initializers/chat-sidebar.js:223]: could call `this.channel.slug` ?

File: plugins/chat/assets/javascripts/discourse/initializers/chat-sidebar.js [hunk 7]
```diff
@@ -57,7 +57,7 @@ export default {
...
```
Comment (jjaffeux) [inline plugins/chat/assets/javascripts/discourse/initializers/chat-sidebar.js:60]: could call `this.channel.slug` ?

File: plugins/chat/assets/javascripts/discourse/services/chat-notification-manager.js [hunk 8]
```diff
@@ -57,7 +57,7 @@ export default class ChatNotificationManager extends Service {
 
   @bind
   _pageChanged(path) {
-    this.set("_inChat", path.startsWith("/chat/channel/"));
+    this.set("_inChat", path.startsWith("/chat/c/"));
```
Comment (jjaffeux) [inline plugins/chat/assets/javascripts/discourse/services/chat-notification-manager.js:60]: I need to change this 😅 Forgot it

File: plugins/chat/assets/javascripts/discourse/services/chat.js [hunk 9]
```diff
@@ -297,8 +297,8 @@ export default class Chat extends Service {
 
       return this.router.transitionTo(
         "chat.channel",
-        channel.id,
         slugifyChannel(channel),
```
Comment (jjaffeux) [inline plugins/chat/assets/javascripts/discourse/services/chat.js:300]: `channel.slugifiedTitle`

**Comment by jjaffeux** [comment]
I think you could investigate rewriting `slug` to `slug || slugifyChannel(this)` when we create the model so we don't have this slightly odd `slugifyTitle` method.

Look at what we do here for example: https://github.com/discourse/discourse/blob/main/app/assets/javascripts/discourse/app/models/badge.js#L66

**Comment by jjaffeux** [comment]
Let me know when you want me to make another review of this

**Comment by romanrizzi** [comment]
@jjaffeux - I made another pass and added the `routeModels` method you suggested. As mentioned in https://github.com/discourse/discourse/pull/18879, it's not possible to rely 100% on `channel.slug` because of DMs. At least, we always access it through the channel model now. cc @martin-brennan because he worked on slugs.

File: plugins/chat/assets/javascripts/discourse/components/reviewable-chat-message.hbs [hunk 10]
```diff
@@ -1,7 +1,7 @@
 <div class="flagged-post-header">
   <LinkTo
     @route="chat.channel"
-    @models={{array this.chatChannel.id this.chatChannel.title}}
+    @models={{array this.chatChannel.title this.chatChannel.id}}
```
Comment (jjaffeux) [inline plugins/chat/assets/javascripts/discourse/components/reviewable-chat-message.hbs:4]: forgot a routeModels?

File: plugins/chat/assets/javascripts/discourse/routes/chat-channel.js [hunk 11]
```diff
@@ -17,11 +16,15 @@ export default class ChatChannelRoute extends DiscourseRoute {
...
     const { channelTitle, messageId } = this.paramsFor(this.routeName);
-    const slug = slugifyChannel(model);
-    if (channelTitle !== slug) {
-      this.router.replaceWith("chat.channel.index", model.id, slug, {
-        queryParams: { messageId },
-      });
+    if (channelTitle !== model.slugifiedTitle) {
+      this.router.replaceWith(
+        "chat.channel.index",
+        model.slugifiedTitle,
```
Comment (jjaffeux) [inline plugins/chat/assets/javascripts/discourse/routes/chat-channel.js:22]: would `...model.routeModels`, works here?

File: plugins/chat/assets/javascripts/discourse/components/reviewable-chat-message.hbs (hunk 10 above)
Comment (romanrizzi) [reply plugins/chat/assets/javascripts/discourse/components/reviewable-chat-message.hbs:4]: Yes!

**Comment by jjaffeux** [comment]
@romanrizzi sadly this is not working for client only redirection, try to post a link to a channel before applying your branch, you will get a onebox linking to a channel with this format: `http://127.0.0.1:4200/chat/channel/7/sporer-lindgren` apply your branch, go to the channel with this onebox, click on it, it will 404, if you refresh page at this point it will correctly redirect because the backend will operate the redirection.

**Comment by romanrizzi** [comment]
@jjaffeux - I added a temporary route to maintain backwards-compatibility. Coming up with a rebaking strategy is going to be hard, so let's keep this around instead and remove it after the 3.1 release. I think it's a good idea because messages will be eventually destroyed.

**Comment by jjaffeux** [comment]
> @jjaffeux - I added a temporary route to maintain backwards-compatibility. Coming up with a rebaking strategy is going to be hard, so let's keep this around instead and remove it after the 3.1 release. I think it's a good idea because messages will be eventually destroyed.

Sounds good, will retry tomorrow, thanks :+1:

**Comment by jjaffeux** [comment]
Great work thanks, looks good to me :+1: Please merge it early in your day if you ca.

**Review by jjaffeux — APPROVED** [review]

**Pull request merged**